menas/ui/service/SchemaService.js

```diff
@@ -103,8 +103,21 @@ var SchemaService = new function() {
       if (xhr.status === 400) {
         let err = "Disabling schema failed. Clear the following dependencies first:\n";
         let oData = JSON.parse(xhr.responseText);
-        for(let ind in oData) {
-          err += "\t - " + oData[ind].name + " (v. " + oData[ind].version + ")";
+
+        let datasets = oData["datasets"];
+        if (datasets.length !== 0) {
+          err += "Datasets:\n";
+          for (let ind in datasets) {
+            err += "- " + datasets[ind].name + " (v. " + datasets[ind].version + ")\n";
+          }
+        }
+
+        let mappingTables = oData["mappingTables"];
```

Comment:
This repeats lines 107-113 almost exactly. Consider combining the blocks in a function.

Reply:
Also, similar code is in MappingTableService.js:175. Looks like this could be a library function.

Reply:
Fair point, need to do a lot of refactoring in these services.